common/coin_support/coin_utils.c

```diff
@@ -456,4 +489,111 @@ void bech32_addr_encode(char *output, char *hrp, uint8_t *address_bytes, uint8_t
   size_t datalen = 0;
   convert_bits(data, &datalen, 5, address_bytes, byte_len, 8, 1);
   bech32_encode(output, hrp, data, datalen);
+}
+
+bool verify_xpub_derivation_path(const uint32_t *path, uint8_t depth) {
+    bool status = false;
+
+    if (depth < 2) return status;
+
+    uint32_t purpose = path[0], coin = path[1];
+
+    switch (coin) {
+        case NEAR:
+            status = near_verify_derivation_path(path, depth);
+            break;
+
+        case SOLANA:
+            status = sol_verify_derivation_path(path, depth);
+            break;
+
+        case LITCOIN:
+        case DOGE:
+        case DASH:              // m/44'/5'  /i'
+        case ETHEREUM:          // m/44'/60' /i'
+            status = (purpose == NON_SEGWIT);
+
+        case BTC_TEST:          // m/44'/1'  /i'
+        case BITCOIN:           // m/44'/0'  /i'
+            status = (purpose == NON_SEGWIT || purpose == NATIVE_SEGWIT);
+            break;
+
+        default:
+            break;
+    }
+
+    return status;
+}
+
+bool verify_receive_derivation_path(const uint32_t *path, uint8_t depth) {
```

Comment:
Are we planning to use this function in future?

Reply:
thanks for reminding. i have to add this in receive address flow.